dclab/rtdc_dataset/writer.py

```diff
@@ -217,7 +217,15 @@ def store_feature(self, feat, data, shape=None):
             else:
                 raise ValueError(f"Bad shape for {feat}! Expeted {shape}, "
                                  + f"but got {data.shape[1:]}!")
-            self.write_ndarray(group=events, name=feat, data=data)
+                
+            # above code ensure that the shape of the (plugin or temp) data 
+            # must be 3 dimensions --> (None, dim1, dim2).
+            # If dim1 and dim2 are greater than 3, it would be considered as 
+            # an image. Otherwise, a ndarray. 
+            if data.shape[1] > 3 and data.shape[2] > 3: 
```

Comment:
I assume this if statement will cause an index error if the data has less than 2 dimensions? 
It would be good to write a test for this situation.

Reply:
Yes, I just added one more line on top of it to check the length of the data.